test/bin/mdx-dune-gen/misc/non-deterministic/dune

```diff
@@ -0,0 +1,52 @@
+(rule
+ (target dune_gen.ml)
+ (action
+  (with-stdout-to
+   %{target}
+   (run ocaml-mdx dune-gen))))
+
+(executable
+ (name dune_gen)
+ (modules dune_gen)
+ (modes byte)
+ (libraries mdx.test))
+
+(rule
+ (target dune-mdx-nondeterministic.deterministic)
+ (deps
+  dune_gen.exe
+  (:input dune-mdx-nondeterministic))
+ (action
+  (with-stdout-to
+   %{target}
+   (run ./dune_gen.exe %{input}))))
+
+(rule
+ (target dune-mdx-nondeterministic.nondeterministic)
+ (deps
+  dune_gen.exe
+  (:input dune-mdx-nondeterministic))
+ (action
+  (setenv
+   MDX_RUN_NON_DETERMINISTIC
+   1
+   (with-stdout-to
+    %{target}
+    (run ./dune_gen.exe %{input})))))
+
+(rule
+ (alias runtest)
+ (action
+  (diff %{dep:dune-mdx-nondeterministic.expected}
+    %{dep:dune-mdx-nondeterministic.deterministic})))
+
+;; make sure the non-deterministic is different from the deterministic
+
+(rule
+ (alias runtest)
+ (action
+  (with-accepted-exit-codes
+   1
+   (ignore-stdout
+    (run diff %{dep:dune-mdx-nondeterministic.expected}
```

Comment:
I'm not completely sure that we can expect the `diff` binary to be available in PATH but can't think of a way to do that with just dune.

Reply:
I was scouting what I can use because I didn't find a way to basically negate `(diff)` so `diff` (and `cmp`) was the best I could find. At least it seems to consistently use 0, 1 and >1 for exit codes between GNU and BSD.

The alternative was shelling out and dealing with `[` or `[[` but that seemed even worse and possibly even less portable.

Fortunately these are just required to run the tests so unless someone wants to run tests on Windows without WSL or MinGW I think it is acceptable for now.

Reply:
An alternative I just thought of is writing a short OCaml helper binary that checks whether two files are different, and given the it can terminate on the first byte that doesn't match without any kind of complex reporting should be quite a simple program.

Reply:
In the future we can always disable that specific test on windows and eventually write a windows version of it.

Reply:
I decided to make an OCaml version of it since the functionality is simple and it doesn't have to be particularly efficient to compare two text files. Thus I could write it in a way where I don't need to redirect stdout or deal with exit codes without having to write anything too complex. In a way I find the build now more understandable since I'm not using `diff` to do the exact opposite of what people use `diff` for.